### DISCUSSION TITLE
What does the value of Outcome::Forward do?

### DISCUSSION THREAD

**Discussion by najtin** [rwf2/Rocket#2060] [Questions]
Hi, the only thing in the documentation i could find was: 
> Contains the value to forward on.

What does that mean? Is the data somehow supplied to the next route?

See https://api.rocket.rs/master/rocket/outcome/enum.Outcome.html and
https://api.rocket.rs/master/rocket/outcome/index.html

**Comment by somehowchris** [comment]
Well it's really good described alongside the example of https://api.rocket.rs/master/rocket/outcome/index.html#forward

**Reply by najtin** [reply]
I had read this document and i don't think it describes it well. It says:
> A forward Outcome<S, E, F>, Forward(F), is returned when a function wants to indicate that the requested processing should be forwarded to the next available processor. Again, the exact meaning depends on the context.

Okay, it depends on the context? Where does it forward to if it is used in a guard?

**Reply by somehowchris** [reply]
>  Instead, the incoming request is forwarded, or passed on to, the next matching route, if any

**Reply by najtin** [reply]
Ah, thanks. The value in the Forward is ignored then, right? Or can the route somehow access the forwarded value?

**Reply by somehowchris** [reply]
Please read the docs

**Reply by somehowchris** [reply]
> Ultimately, if there are no non-forwarding routes, forwarded requests are handled by the 404 catcher. Similar to Failures, users can catch Forwards by requesting a type of Option\<S\>.  If an Outcome is a Forward, the Option will be None.

**Reply by najtin** [reply]
I read this page up and down multiple times before asking the question here, and i don't find the answer there. Yes, Forward can be catched, but the question is what happens to the value of type F supplied during the constructions of  Forward? An Outcome has type Outcome<S, E, F> where S is the type of the data supplied to Outcome::Success and F is the type of the data supplied to Outcome::Forward.
`Outcome::Forward(something of type F)` What is `something of type F` used for? Can the next route somehow access it? If not, what is it for? Is it the request that should be inside the forward`Outcome::Forward(request)`?
Maybe i will take a look into the source code tomorrow and try to figure it out.

**Reply by somehowchris** [reply]
You might wanna load a guard within a guard and then check on it's outcome but I'm not aware of any feature letting you have a look at the outcome of a previous route, you may wanna check out `local_cache` If you need to save something within the scope of the request

**Reply by the10thWiz** [reply]
It depends on the use case. In `FromData`, the forward variant is used to pass the data on to the next handler.

In `FromRequest`, the forward variant has no data (it's literally `()`).

In general, my understand is that `F` is only used when needed, otherwise it doesn't actually contain any information.
